src/libmeasurement_kit/ooni/telegram.cpp

```diff
@@ -34,7 +34,7 @@ static void tcp_many(const std::vector<std::string> ip_ports, SharedPtr<Entry> e
                 logger->info("telegram: success TCP connecting to %s:%d",
                     ip.c_str(), port);
                 result["status"]["success"] = true;
-                (*entry)["telegram_tcp_blocking"] = true;
+                (*entry)["telegram_tcp_blocking"] = false;
```

Comment:
This is not the only logic bug. You are also re-initialising `telegram_tcp_blocking` to `true` in both `http_many` and `tcp_many`. Doing so means that the condition "consider it to work if at least one is working" will not be met in many cases.

Reply:
FTR, it seems that now this has been fixed!